### DISCUSSION TITLE
Using ClangD extension with monaco-vscode-api.

### DISCUSSION THREAD

**Discussion by navpreet032** [CodinGame/monaco-vscode-api#203] [Q&A] [answered]
How can I use the ClangD extension with the monaco-vscode-api? I encountered an error **'Cannot read properties of undefined (reading 'toString')'** when attempting to import the .vsix file. 

`export default defineConfig({
  plugins: [react(),
    vsixPlugin()], 
})`

`import './vscode-clangd-0.1.23.vsix';`

**Comment by CGNonofr** [comment]
I'm not able to reproduce you error, do you have the full stack? are you able to reproduce in the demo of this repository?

Note that even without error, it won't do much as the extension is not web-compatible

You'll need to use the ext-host server to run the extension on a server to make it fully work

**Reply by navpreet032** [reply]
Here is the repo: https://github.com/navpreet032/MonacoDemo
All the details are present in the repo

**Reply by CGNonofr** [reply]
You app is almost empty, there is no service initialization

> Does vsCode also uses ext-host server for clangD extension?

This library bundles VSCode browser code, the code that is used by https://vscode.dev/. The clangd extension can't be run in vscode.dev either and you'll need the ext-host server as well

> if the extension can works fine in VsCode.Then can it work on monaco also?

Yes if it works in vscode.dev

**Reply by navpreet032** [reply]
"no service initialization" i did't got it can you explain it and provide an example code ?

**Reply by CGNonofr** [reply, marked as answer]
I mean https://github.com/CodinGame/monaco-vscode-api/blob/52c94ba8eef8b452b0610007f120dcdc35fc6085/demo/src/setup.ts#L72-L110
